### ISSUE TITLE
Default config does not pick up `.d.ts` files

### ISSUE THREAD

**Issue by chriskrycho** [emberwatch/ember-cli-typescript#15]
Just what it says on the tin. We tried to add a definition file, it failed to compile; we changed the extension to just `.ts` instead of `.d.ts` and it worked.

Probably just what the tree is configured to snag, I'm guessing. Will take a look down the line.

**Comment by chriskrycho** [comment]
Confirm: current implementation [only picks up `ts` as the extension](https://github.com/emberwatch/ember-cli-typescript/blob/master/lib/typescript-preprocessor.js#L10-L15).

**Comment by chriskrycho** [comment]
The problem was actually in the `includeExtensions` argument to the `TsPreprocessor` constructor.

**Comment by chriskrycho** [comment]
An alternative approach to consider: stop [setting weird values in the tsconfig](https://github.com/emberwatch/ember-cli-typescript/issues/21) and see what happens.

**Comment by leepfrog** [comment]
Here's my `tsconfig.json` -- it seems to be working with new types defined correctly with the `app/config` folder (eg: `app/config/<module_name>.d.ts`)

```
{
  "compilerOptions": {
    "target": "ES6",
    "allowJs": true,
    "moduleResolution": "node",
    "noEmitOnError": false,
    "noEmit": true,
    "baseUrl": ".",
    "paths": {
      "mycoolappname/*": ["app/*"],
      "mycoolappname/tests/*": ["tests/*"],
      "mycoolappname/mirage/*": ["mirage/*"]
    }
  },
  "include": [
    "app/**/*",
    "tests/**/*",
    "mirage/**/*"
  ]
}
```

an example of defining support for a module that defines globals:
`app/config/ember-testing.d.ts`
```typescript
declare var visit: (path: string) => PromiseConstructor;
declare var triggerEvent: (selector: string | JQuery, event: string) => PromiseConstructor;
declare var click: (selector: string | JQuery) => PromiseConstructor;
declare var find: (selector: string | JQuery) => JQuery;
```

an example of defining support for a module that you import from:
`app/config/qunit.d.ts`
see: https://gist.github.com/leepfrog/0c08bd20ab54a0ba957a4504e87703cf

an example of defining support for something you're going to only use in TS (eg: I converted all of my /mirage folder to TS):
`app/config/ember-cli-mirage.d.ts`
```
declare namespace Mirage {
  export namespace Model {
    export interface Base {
      id: string;
    }

    // I'll move this into app/models/my-cool-model.ts when it's no longer .js (it would be nice to have ember-cli-typescript auto-define these modules in appropriate namespaces if that makes sense?)
    export interface MyCoolModel extends Base {

    }
    // ....
  }
}

declare var server: any;  
```

**Comment by chriskrycho** [comment]
Thanks, @leepfrog! That's a super helpful bit of data. It confirms that the problem was indeed resolved with #22 (as I thought it would be).
